tuf/client/client.go

```diff
@@ -111,10 +110,6 @@ func (c Client) checkRoot() error {
 		return fmt.Errorf("Cached root sha256 did not match snapshot root sha256")
```

Comment:
`bytes.Equal` will only return true in the case that the 2 slices are identical, including length (the documentation explicitly confirms the length is checked).